### DISCUSSION TITLE
Docker SharedFolderSync not possible, compose doesnt work

### DISCUSSION THREAD

**Discussion by Routhgen** [abraunegg/onedrive#1780] [Q&A] [answered]
Hello,

I would say this is the best option for OneDrive Bussiness while on Linux - havent found any better alternative.

I just followed all the docker steps.
Out of the box it works fine, its synced - everything is fine.

1.
However - compose doesnt work and it argues about used volume but not volume section.
So I added
volumes:
    onedrive_conf:

Thats fine but even step 3. First run was performed and container stopped the docker-compose up is trying to authenticate again because its not started as onedrive but as onedrive_onedrive_1 container

May I ask whats the issue? Nobody really wants to use docker run for service usage. For this purpose we have compose.

2.
Also another point is that when sync_business_shared_folders is uncommented and set to "true" together with listed available shared folders (one organization) & putting one to the new file within conf volume image.
The whole process is failing like this when trying to run resync as suggested in the docs:

```docker container run -e ONEDRIVE_RESYNC=1 -v onedrive_conf:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" driveone/onedrive:latest
++ stat /onedrive/data -c %u
+ : 1000
++ stat /onedrive/data -c %g
+ : 1000
++ getent group 1000
+ odgroup=
+ odgroup=onedrive
+ groupadd onedrive -g 1000
++ getent passwd 1000
+ oduser=
+ oduser=onedrive
+ useradd -m onedrive -u 1000 -g 1000
+ chown onedrive:onedrive /onedrive/ /onedrive/conf
+ ARGS=(--monitor --confdir /onedrive/conf --syncdir /onedrive/data)
+ '[' 0 == 1 ']'
+ '[' 0 == 1 ']'
+ '[' 0 == 1 ']'
+ '[' 1 == 1 ']'
+ echo '# We are performing a --resync'
+ ARGS=(--resync ${ARGS[@]})
# We are performing a --resync
+ '[' 0 == 1 ']'
+ '[' 0 == 1 ']'
+ '[' 0 -gt 0 ']'
+ exec gosu onedrive /usr/local/bin/onedrive --resync --monitor --confdir /onedrive/conf --syncdir /onedrive/data
Configuration file successfully loaded
config file has been updated, checking if --resync needed
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Initializing monitor ...
OneDrive monitor interval (seconds): 300
Retry sync count: 1: Key not found: remaining
Retry sync count: 2: Key not found: remaining
Giving up on sync after three attempts: Key not found: remaining
std.json.JSONException@std/json.d(641): Key not found: remaining
----------------
/usr/include/dmd/phobos/std/exception.d:516 _D3std9exception__T7bailOutHTCQBc4json13JSONExceptionZQBjFNaNfAyamMAxaZNn [0x69da1b]
??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException).enforce!(inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x7679d8]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(immutable(char)[]) [0x7137dd]
src/sync.d:1178 void sync.SyncEngine.applyDifferences(immutable(char)[], const(char)[], bool) [0x6d56ae]
src/sync.d:733 void sync.SyncEngine.applyDifferences(bool) [0x6d379c]
src/main.d:1389 void main.performSync(sync.SyncEngine, immutable(char)[], bool, bool, bool, long, bool, bool, bool, bool, monitor.Monitor) [0x6c038a]
src/main.d:1180 _Dmain [0x6bf269]
std.net.curl.CurlException@std/net/curl.d(4402): A libcurl function was given a bad argument on handle null
----------------
??:? _D3std9exception__T7bailOutHTCQBc3net4curl13CurlExceptionZQBnFNaNfAyamMAxaZNn [0x767dde]
??:? pure @safe bool std.exception.enforce!(std.net.curl.CurlException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x767d4b]
??:? void std.net.curl.Curl._check(int) [0x734d77]
??:? void std.net.curl.Curl.clear(etc.c.curl.CurlOption) [0x735126]
??:? void std.net.curl.HTTP.clearRequestHeaders() [0x733e7b]
src/onedrive.d:366 void onedrive.OneDriveApi.shutdown() [0x6c434c]
src/main.d:63 _Dmain [0x6bf8d6]
```
One available shared folder in this file
`/var/lib/docker/volumes/onedrive_conf/_data/business_shared_folders`
Enabled via sync_business_shared_folders = "true" in 
`/var/lib/docker/volumes/onedrive_conf/_data/config`

```
docker volume inspect onedrive_conf
[
    {
        "CreatedAt": "2022-01-04T14:51:39+01:00",
        "Driver": "local",
        "Labels": {},
        "Mountpoint": "/var/lib/docker/volumes/onedrive_conf/_data",
        "Name": "onedrive_conf",
        "Options": {},
        "Scope": "local"
    }
]
```

The shared folders sync is mandatory for us - can you please suggest what Im doing wrong? Or is there any other issue so I have to leave docker & install it manually under Fedora?
Thanks.

Docker is highly preferable due to its availability everywhere & easy to reproduce.

Regards,
Zbynek

**Comment by abraunegg** [comment, marked as answer]
@Routhgen 
> Docker SharedFolderSync not possible, compose doesnt work

This is because Docker changed their integration policies so that 'free' Docker accounts cannot 'push' new releases to Docker - so you are building against an obsolete client version that is hit by this [issue](https://github.com/abraunegg/onedrive/issues/1712) which was fixed by this [PR](https://github.com/abraunegg/onedrive/pull/1713).

Your Docker compose will work when you build your Docker container from source from GitHub

Relying on Docker Hub images for OneDrive for Business is fruitless unless Docker accepts the submission I made for access due to providing an opensource solution.

**Reply by Routhgen** [reply]
Thanks for the feedback. I already tried to use standard installation.
It works but again its unable to sync shared folders, same error:
Retry sync count: 1: Key not found: remaining
Isnt there a bug? Looks like key remaining is expected but not given.
Using latest available version for Fedora onedrive v2.4.13
Do you know about such a issue which is already resolved so Im suppose to compile the latest?

**Comment by abraunegg** [comment]
@Routhgen 
> Using latest available version for Fedora onedrive v2.4.13

This version is obsolete and not supported any more. Please read the [install](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md) document very carefully.
